druid/src/widget/image.rs

```diff
@@ -212,3 +212,142 @@ impl Default for ImageData {
+                // the padding collor and the middle rows will not have any padding
+
+                // check that the middle row 400 pix wide is 200 black then 200 wight
+                let expecting: Vec<u8> = [
+                    vec![0, 0, 0, 255].repeat(200),
+                    vec![255, 255, 255, 255].repeat(200),
+                ]
+                .concat();
+                assert_eq!(raw_pixels[400 * 300 * 4..400 * 301 * 4], expecting[..]);
+
+                // check that the hole of the last 100 rows are all the back ground color.
+                let expecting: Vec<u8> = vec![41, 41, 41, 255].repeat(400 * 100);
+                assert_eq!(
+                    raw_pixels[400 * 600 * 4 - 4 * 400 * 100..400 * 600 * 4],
+                    expecting[..]
+                );
+            },
+        )
+    }
+    #[test]
+    fn wide_paint() {
+        use crate::{tests::harness::Harness, WidgetId};
+        let _id_1 = WidgetId::next();
+        let image_data = ImageData {
+            pixels: vec![255, 255, 255, 0, 0, 0, 0, 0, 0, 255, 255, 255],
+            x_pixels: 2,
+            y_pixels: 2,
+            format: ImageFormat::Rgb,
+        };
+
+        let image_widget =
+            Image::new(image_data).interpolation_mode(InterpolationMode::NearestNeighbor);
+
+        Harness::create_with_render(
+            true,
+            image_widget,
+            Size::new(600., 400.),
+            |harness| {
+                harness.send_initial_events();
+                harness.just_layout();
+                harness.paint();
+            },
+            |target| {
+                let raw_pixels = target.into_raw();
+                assert_eq!(raw_pixels.len(), 400 * 600 * 4);
+
+                // being a wide widget every row will have some padding at the start and end
+                // the last row will be like this too and there will be now padding rows at the end.
+
+                // a middle row of 600 pixels is 100 padding 200 black, 200 wide and then 100 padding
+                let expecting: Vec<u8> = [
+                    vec![41, 41, 41, 255].repeat(100),
+                    vec![255, 255, 255, 255].repeat(200),
+                    vec![0, 0, 0, 255].repeat(200),
+                    vec![41, 41, 41, 255].repeat(100),
+                ]
+                .concat();
+                assert_eq!(raw_pixels[199 * 600 * 4..200 * 600 * 4], expecting[..]);
+
+                // the final row of 600 pixels is 100 padding 200 black, 200 wide and then 100 padding
+                let expecting: Vec<u8> = [
+                    vec![41, 41, 41, 255].repeat(100),
+                    vec![0, 0, 0, 255].repeat(200),
+                    vec![255, 255, 255, 255].repeat(200),
+                    vec![41, 41, 41, 255].repeat(100),
+                ]
+                .concat();
+                assert_eq!(raw_pixels[399 * 600 * 4..400 * 600 * 4], expecting[..]);
```

Comment:
@raphlinus I think you did a lot of the piet logic, can you think of a nicer way to do this. I have made these tests so that they should be very stable but if people want to use this for more complex widgets then crafting a expected slice could get quite trouble some especially if they are not to be pretty fragile.

I was thinking of maybe using piet to create a similar image and then comparing the hole things but I'm not sure that would be very helpful.

Maybe for another MR but do you have any ideas?